### DISCUSSION TITLE
Fix Terminal/Logbook "Welcome Message" sending multiple times.

### DISCUSSION THREAD

**Discussion by Demyxa** [FakeFishGames/Barotrauma#6307] [Ideas and suggestions]
Terminals and Logbooks 

have a "Welcome Message" Parameter, allowing for a String to be sent to the Terminal as a... well, welcome message.

This however appears bugged and the welcome message is sent several times when "initializing" the sub several times.
**Basically:** Change it so the welcome message only ever appears once.

![20210618110045_1](https://user-images.githubusercontent.com/48883376/125604436-a4b6e00a-f02a-4096-864f-cbc60e7aec84.jpg)

![20210704161135_1](https://user-images.githubusercontent.com/48883376/125604526-a5c4bb7f-c6ea-4fd7-a976-f0f01fe3dd5e.jpg)

**Comment by Ek-Videogames** [comment]
Possible known bug / related bug.
 
 — 06/03/2021
> in a patch they changed the format of terminal messages, a weird side effect is that terminals saved from previous versions display their welcome message twice
because the old formatting used `msg0` to denote welcome message, but now its `welcomemessage`
but instead of deleting the old one they just added the new one - and the obsolete method still works for compatibility reasons
so the game just prints the same message twice
